src/components/PolkadotIdenticon/utils.ts

```diff
@@ -0,0 +1,147 @@
+}
+
+const getCircleXY = () => {
+  const { r, r3o4, ro2, ro4, rroot3o2, rroot3o4 } = getRotation()
+
+  return [
+    [OUTER_CIRCLE_POSITION, OUTER_CIRCLE_POSITION - r],
+    [OUTER_CIRCLE_POSITION, OUTER_CIRCLE_POSITION - ro2],
+    [OUTER_CIRCLE_POSITION - rroot3o4, OUTER_CIRCLE_POSITION - r3o4],
+    [OUTER_CIRCLE_POSITION - rroot3o2, OUTER_CIRCLE_POSITION - ro2],
+    [OUTER_CIRCLE_POSITION - rroot3o4, OUTER_CIRCLE_POSITION - ro4],
+    [OUTER_CIRCLE_POSITION - rroot3o2, OUTER_CIRCLE_POSITION],
+    [OUTER_CIRCLE_POSITION - rroot3o2, OUTER_CIRCLE_POSITION + ro2],
+    [OUTER_CIRCLE_POSITION - rroot3o4, OUTER_CIRCLE_POSITION + ro4],
+    [OUTER_CIRCLE_POSITION - rroot3o4, OUTER_CIRCLE_POSITION + r3o4],
+    [OUTER_CIRCLE_POSITION, OUTER_CIRCLE_POSITION + r],
+    [OUTER_CIRCLE_POSITION, OUTER_CIRCLE_POSITION + ro2],
+    [OUTER_CIRCLE_POSITION + rroot3o4, OUTER_CIRCLE_POSITION + r3o4],
+    [OUTER_CIRCLE_POSITION + rroot3o2, OUTER_CIRCLE_POSITION + ro2],
+    [OUTER_CIRCLE_POSITION + rroot3o4, OUTER_CIRCLE_POSITION + ro4],
+    [OUTER_CIRCLE_POSITION + rroot3o2, OUTER_CIRCLE_POSITION],
+    [OUTER_CIRCLE_POSITION + rroot3o2, OUTER_CIRCLE_POSITION - ro2],
+    [OUTER_CIRCLE_POSITION + rroot3o4, OUTER_CIRCLE_POSITION - ro4],
+    [OUTER_CIRCLE_POSITION + rroot3o4, OUTER_CIRCLE_POSITION - r3o4],
+    [OUTER_CIRCLE_POSITION, OUTER_CIRCLE_POSITION],
+  ]
+}
+
+const findScheme = (d: number) => {
+  let cum = 0
+  const schema = Object.values(SCHEMA).find((schema): boolean => {
+    cum += schema.freq
+
+    return d < cum
+  })
+
+  if (!schema) {
+    throw new Error('Unable to find schema')
+  }
+
+  return schema
+}
+
+const addressToId = (address: string) => {
+  return blake2(decodeAddress(address)).map((x, i): number => (x + 256 - ZERO[i]) % 256)
+}
+
+const getColors = (address: string) => {
+  const total = Object.values(SCHEMA)
+    .map((s): number => s.freq)
+    .reduce((a, b): number => a + b)
+  const id = addressToId(address)
+  const d = Math.floor((id[30] + id[31] * 256) % total)
+  const rot = (id[28] % 6) * 3
+  const sat = (Math.floor((id[29] * 70) / 256 + 26) % 80) + 30
+  const scheme = findScheme(d)
+  const palette = Array.from(id).map((x, i): string => {
+    const b = (x + (i % 28) * 58) % 256
+
+    if (b === 0) {
+      return '#444'
+    } else if (b === 255) {
+      return 'transparent'
+    }
+
+    const h = Math.floor(((b % 64) * 360) / 64)
+    const l = [53, 15, 35, 75][Math.floor(b / 64)]
+
+    return `hsl(${h}, ${sat}%, ${l}%)`
+  })
+
+  return scheme.colors.map((_, i): string => palette[scheme.colors[i < 18 ? (i + rot) % 18 : 18]])
+}
+
+/**
+ * @description Generate a array of the circles that make up an identicon
+ */
+export const polkadotIcon = (address: string) => {
+  const colors = getColors(address)
+
+  return [OUTER_CIRCLE].concat(
+    getCircleXY().map(
+      ([cx, cy], index): Circle => ({
+        cx,
+        cy,
+        fill: colors[index],
+        r: DOT_RADIUS,
+      })
+    )
+  )
+}
```

Comment:
I think you shouldn't bother reviewing this file too much. It's just copied code from [here] with my very small adjustments (https://github.com/polkadot-js/ui/blob/8bdfd18e1fbaaa0f3b6a6a098a6705370b3ae306/packages/ui-shared/src/icons/polkadot.ts)